js/apps/admin-ui/src/i18n/OverridesBackend.ts

```diff
@@ -0,0 +1,160 @@
+    }
+  }
+
+  #applyOverrides(
+    namespace: string,
+    data: ResourceKey,
+    overrides: ParsedOverrides
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (typeof data === "string" || !overrides[namespace]) {
+      return data;
+    }
+
+    // Ensure we are operating on a cloned data structure to prevent in place mutations.
+    const target = structuredClone(data);
+
+    for (const [path, value] of Object.entries(overrides[namespace])) {
+      this.#applyOverride(target, path, value);
+    }
+
+    return target;
+  }
+
+  /** Applies an override by converting path segments denoted with a key separator as nested objects and merging the result. */
+  #applyOverride(target: Record<string, any>, path: string, value: string) {
+    const trail = path.split(KEY_SEPARATOR);
+    let pointer = target;
+
+    trail.forEach((segment, index) => {
+      const isLast = index === trail.length - 1;
+      pointer = pointer[segment] = isLast ? value : pointer[segment] ?? {};
+    });
+  }
+
+  #loadOverrides(languages?: string | string[]) {
+    const locale = this.#determineLocale(languages);
+    const cachedOverrides = this.#overridesCache.get(locale);
+
+    if (cachedOverrides) {
+      return cachedOverrides;
+    }
+
+    const overrides = adminClient.realms
+      .getRealmLocalizationTexts({
+        realm: adminClient.realmName,
+        selectedLocale: locale,
+      })
+      .then((data) => this.#parseOverrides(data));
+
+    this.#overridesCache.set(locale, overrides);
+
+    // Evict cached request on failure.
+    overrides.catch((error) => {
+      this.#overridesCache.delete(locale);
+      return Promise.reject(error);
+    });
+
+    return overrides;
+  }
+
+  #parseOverrides(data: Record<string, string>) {
+    const parsed: ParsedOverrides = {};
+
+    for (const [path, value] of Object.entries(data)) {
+      const parts = path.split(NAMESPACE_SEPARATOR);
+      const namespace = parts.length > 1 ? parts[0] : DEFAULT_NAMESPACE;
+      const key = parts.length > 1 ? parts[1] : parts[0];
+
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (!parsed[namespace]) {
+        parsed[namespace] = {};
+      }
+
+      parsed[namespace][key] = value;
+    }
+
+    return parsed;
+  }
+
+  #determineLocale(languages?: string | string[]) {
+    if (typeof languages === "string") {
+      return languages;
+    }
+
+    return languages?.[0] ?? DEFAULT_LOCALE;
+  }
+
+  #determineNamespace(namespaces?: string | string[]) {
+    if (typeof namespaces === "string") {
+      return namespaces;
+    }
+
+    return namespaces?.[0] ?? DEFAULT_NAMESPACE;
+  }
+
+  #loadUrlPromisified(
+    url: string,
+    languages?: string | string[],
+    namespaces?: string | string[]
+  ) {
+    return new Promise<ResourceKey>((resolve, reject) => {
+      const callback: ReadCallback = (error, data) => {
+        if (error) {
+          return reject(error);
```

Comment:
I would suggest to return `null` (or some kind of "null-object") instead of a failure, and adjusting the method `#applyOverrides` to just return the realm localization texts in this case.
Otherwise it is not possible to use realm localization texts when there are no translation files provided in the Admin UI json files. (Actually, it's a use case for some of our customers, to provide translations via realm localization, when there are no translation files for a required language.)

Reply:
I see your use-case for this, but perhaps we should handle this in a separate issue/PR. Reason being that this behavior is already present in the standard `HttpBackend` we're overriding here.

I'd also like to discuss this with the larger team, where an issue would help start that discussion.